src/Breakpoint.php

```diff
@@ -2,80 +2,103 @@
 
 namespace Spatie\ResponsiveImages;
 
+use Exception;
 use Illuminate\Contracts\Support\Arrayable;
 use Illuminate\Support\Collection;
 use Illuminate\Support\Facades\Config;
 use League\Flysystem\FilesystemException;
-use Spatie\ResponsiveImages\Jobs\GenerateImageJob;
 use Statamic\Contracts\Assets\Asset;
 use Statamic\Facades\Blink;
 use Statamic\Facades\Glide as GlideManager;
 use Statamic\Imaging\ImageGenerator;
 use Statamic\Support\Str;
 
+/**
+ * @property-read \Statamic\Assets\Asset $asset
+ * @property-read string $label
+ * @property-read int $minWidth
+ * @property-read array $params
+ * @property-read string $widthUnit
+ */
 class Breakpoint implements Arrayable
 {
-    /** @var \Statamic\Assets\Asset */
-    public $asset;
+    /** @var Asset */
+    public Asset $asset;
 
     /** @var string */
-    public $label;
+    public string $label;
 
-    /** @var ?int */
-    public $value;
+    /**
+     * @var int The minimum width of when the breakpoint starts
+     */
+    public int $minWidth;
 
     /** @var array */
-    public $parameters;
-
-    /** @var float */
-    public $ratio;
+    public array $params;
 
     /** @var string */
-    public $unit;
+    public string $widthUnit;
+
+    /** @var Collection<Source> */
+    private Collection $sources;
 
-    public function __construct(Asset $asset, string $label, int $value, array $parameters)
+    public function __construct(Asset $asset, string $label, int $breakpointMinWidth, array $breakpointParams)
     {
         $this->asset = $asset;
         $this->label = $label;
-        $this->value = $value;
-        $this->parameters = $parameters;
-        $this->ratio = $this->parameters['ratio'] ?? $this->asset->width() / $this->asset->height();
+        $this->minWidth = $breakpointMinWidth;
+        $this->params = $breakpointParams;
+        $this->widthUnit = config('statamic.responsive-images.breakpoint_unit', 'px');
+    }
 
-        $this->unit = config('statamic.responsive-images.breakpoint_unit', 'px');
+    public function __set($name, $value): void
+    {
+        throw new Exception(sprintf('Cannot modify property %s', $name));
     }
 
-    public function getMediaString(): string
+    /**
+     * @return Collection<Source>
+     */
+    public function sources(): Collection
     {
-        if (! $this->value) {
-            return '';
+        if (isset($this->sources)) {
+            return $this->sources;
         }
 
-        return "(min-width: {$this->value}{$this->unit})";
-    }
+        $formats = collect(['avif', 'webp', 'original']);
```

Comment:
So far this is how addon has worked: if you provided a JPG/PNG/WEBP etc. as source asset, then the default "srcSet" would Glide manipulate to the same format as the original asset was. This is default Glide behavior if you do not specify a format.

Responsive images addon config has option to toggle `webp` and `avif` formats, and technically you can change default quality amount for each format too.

The `original` format has been introduced here, and here's few key notes as to why:
1. We have to deal with source image being any possible image format that Glide/Intervention supports, as such we would have to add format toggles for each image format. Having just one `original` seems simpler. Nowadays we really only care for `webp` and `avif` as their support becomes more robust as time goes.
2. This `original` format cannot be exchanged to just `jpg` because in situations where source asset is a transparent image and that transparency needs to be maintained. Both WEBP and AVIF support transparency. If we force JPG, then transparency will become borked.

For now `original` is not publicly available to devs through config or params. But if you really want to disable original sources, then you can through `DimensionCalculator`. If we are introducing breaking changes, we could add `original` image format to the config too, but I decided against it for now as this PR is becoming too big and would be out of scope technically, trying to remain lean here.